### PR TITLE
admin_user & file_copy: Make paths relative to the manifest

### PR DESCRIPTION
**Pull request by nbraud**
This is a followup on #331, since this specific patch was dropped.

@andsens I agree that paths relative to the manifest would be best.
However, while implementing this I ran into trouble: there is no way for `validate_manifest` to access `manifest.path`.
Moreover, the `file_copy` plugin also allows relative paths, and has the same issue (pathes end up being relative to the working directory).
I suggest that we pass the entire `manifest` object to `validate_manifest`, rather than `manifest.data`, `manifest.schema_validator` and `manifest.validation_error`.

### CODE REVIEW

**Comment by nbraud** [comment]
@andsens This is my proposed solution:
- pass the whole manifest to `validate_manifest()`;
- make paths relatives to the manifest (in `admin_user` and `file_copy`), and validate them properly.

Note that the `file_copy` change is breaking: the documentation previously asserted that paths were relative to the “current directory”.

**Comment by andsens** [comment]
Nice work :-)
I think however a better idea would be to add those validations as tasks, there you have full access to everything that is needed. It might be a good idea to actually have a new "validation" phase before the "preparation" phase, which checks all the external dependencies of the manifest.

**Comment by nbraud** [comment]
@andsens Updated, and I sneaked in a bugfix

File: bootstrapvz/common/tools.py [hunk 1]
```diff
@@ -133,3 +133,12 @@ def copy_tree(from_path, to_path):
         for path in files:
             copy(os.path.join(abs_prefix, path),
                  os.path.join(to_path, prefix, path))
+
+
+def rel_path(base, path):
```
Comment (andsens) [inline bootstrapvz/common/tools.py]: There's actually no need for this one, check this out:
```
$ python
>>> import os
>>> os.path.join('/some/absolute/path', '/another/absolute/path')
'/another/absolute/path'
```
:-)

**Comment by andsens** [comment]
I added a small comment, otherwise this looks great!

**Comment by nbraud** [comment]
Thanks

**Comment by andsens** [comment]
> However, while implementing this I ran into trouble: there is no way for validate_manifest to access manifest.path.

Oh crap. There's a reason for this and I totally forgot about it during this entire PR. You can do remote bootstrapping where the manifest is actually located on another host! I actually have no clue how paths would have worked previously at all.

We definitely need this PR, it's awesome work. It's late though and I need some time and a clear mind to think through the consequences before merging, cool?

**Comment by nbraud** [comment]
@andsens No problem, we aren't on a deadline anyhow.

**Comment by andsens** [comment]
OK. Thought about it. Right now, this would result in an error when using remote bootstrapping. I'll fix it when I get to it. Merging :-)

**Pull request merged**